src/App/MeasureManager.cpp

```diff
@@ -0,0 +1,180 @@
+ *   Lesser General Public License for more details.                       *
+ *                                                                         *
+ *   You should have received a copy of the GNU Lesser General Public      *
+ *   License along with FreeCAD. If not, see                               *
+ *   <https://www.gnu.org/licenses/>.                                      *
+ *                                                                         *
+ **************************************************************************/
+
+#include <Base/Console.h>
+#include <Base/Interpreter.h>
+#include <Base/VectorPy.h>
+#include <App/Document.h>
+
+#include "MeasureManager.h"
+
+namespace App {
+
+    std::vector<MeasureHandler> MeasureManager::_mMeasureHandlers;
+    std::vector<MeasureType*> MeasureManager::_mMeasureTypes;
+
+    MeasureManager::MeasureManager()
+    {
+        // Constructor implementation
+    }
+
+
+    void MeasureManager::addMeasureHandler(const char* module, MeasureTypeMethod typeCb) {
+        auto item = new MeasureHandler{module, typeCb};
+        _mMeasureHandlers.push_back(*item);
+    }
+
+    bool MeasureManager::hasMeasureHandler(const char* module) {
+        for(MeasureHandler& handler : _mMeasureHandlers) {
+            if (strcmp(handler.module.c_str(), module) == 0) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    MeasureHandler MeasureManager::getMeasureHandler(const char* module) {
+        for(MeasureHandler handler : _mMeasureHandlers) {
+            if (!strcmp(handler.module.c_str(), module)) {
+                return handler;
+            }
+        }
+
+        MeasureHandler empty;
+        return empty;
+    }
+
+    void MeasureManager::addMeasureType(MeasureType* measureType) {
+        _mMeasureTypes.push_back(measureType);
+    }
+
+    void MeasureManager::addMeasureType(std::string id, std::string label, std::string measureObj, MeasureValidateMethod validatorCb, MeasurePrioritizeMethod prioritizeCb) {
+        MeasureType* mType = new MeasureType{id, label, measureObj, validatorCb, prioritizeCb, false, nullptr};
+        _mMeasureTypes.push_back(mType);
+    }
+
+    void MeasureManager::addMeasureType(const char* id, const char* label, const char* measureObj, MeasureValidateMethod validatorCb, MeasurePrioritizeMethod prioritizeCb) {
+        addMeasureType(std::string(id), std::string(label), std::string(measureObj), validatorCb, prioritizeCb);
+    }
+
+    const std::vector<MeasureType*> MeasureManager::getMeasureTypes() {
+        return _mMeasureTypes;
+    }
+
+
+    Py::Tuple MeasureManager::getSelectionPy(const App::MeasureSelection& selection) {
+        // Convert selection to python list
+        Py::Tuple selectionPy(selection.size());
+
+        int i = 0;
+        for (auto it : selection) {
+
+            Py::Dict sel;
+            sel.setItem("object", Py::asObject(it.object.getObject()->getPyObject()));
```

Comment:
The selection is passed around with the SubObjectT type. However SubObjectT doesn't seem to have a corresponding python type, therefore a DocumentObjectPy is passed here which might not be ideal.